redis-utils/src/main/java/org/ballerinalang/redis/actions/connection/Ping.java

```diff
@@ -27,31 +28,23 @@
 import org.ballerinalang.natives.annotations.Receiver;
 import org.ballerinalang.redis.Constants;
 import org.ballerinalang.redis.RedisDataSource;
-import org.ballerinalang.redis.RedisDataSourceUtils;
 import org.ballerinalang.redis.actions.AbstractRedisAction;
+import org.ballerinalang.util.exceptions.BallerinaException;
 
 /**
  * {@code {@link Ping}} Maps with "PING" operation of Redis.
  *
  * @since 0.5.0
  */
-@BallerinaFunction(orgName = "wso2",
-                   packageName = "redis:0.0.0",
-                   functionName = "ping",
-                   receiver = @Receiver(type = TypeKind.OBJECT,
-                                        structType = Constants.CLIENT))
 public class Ping extends AbstractRedisAction {
 
-    @Override
-    public void execute(Context context) {
-        BMap<String, BValue> bConnector = (BMap<String, BValue>) context.getRefArgument(0);
-        RedisDataSource redisDataSource = (RedisDataSource) bConnector.getNativeData(Constants.CLIENT);
-
-        BString result = ping(redisDataSource);
+    public static void ping(HandleValue redisDataSourceHandleValue) {
+        RedisDataSource redisDataSource = (RedisDataSource) redisDataSourceHandleValue.getValue();
         try {
-            context.setReturnValues(result);
+            ping(redisDataSource);
         } catch (Throwable e) {
-            context.setReturnValues(RedisDataSourceUtils.getRedisConnectorError(context, e));
+            // Todo throw proper exception
+            throw new BallerinaException("fail");
```

Comment:
Instead of throwing BallerinaException, try using ErrorValue(BallerinaErrors.createError(reason, message)). This is also a runtime exception.

Reply:
Updated in 
c0668f3